### ISSUE TITLE
Consider to implement a generic "constrain tabbing" functionality

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#4621]
This was mentioned here and there in several other issues but, as far as I know, a specific issue hasn't been created yet. So, there it is.

Some Gutenberg UI components work, more or less, like a "modal". They open a floating panel that is visually placed close to the control that opened it, but actually they're placed in the source very far from it.

There are now a few components to set initial focus, move focus back to the "opener" control, get all the tabbable and focusable element, etc. and they're all very useful components to handle focus in a proper way. However, as far as I know, there's no generic component to constrain tabbing yet. 

Some components implement their individual way to constrain tabbing, but other components would need the same feature. Consider for example the Contente Structure `Popover`:

<img width="412" alt="screen shot 2018-01-21 at 18 19 51" src="https://user-images.githubusercontent.com/1682452/35196902-a913295e-fed8-11e7-8462-ea500fa81be8.png">

When it opens, it's possible to tab through its focusable elements. However, it's also possible to "tab away" from it and, since it's placed at the end of the source, tabbing away from it leads users to the end of the page. Basically, the whole interfaces gets "skipped".

This kind of components should be treated like a sort of "modal" and, at the very least, tabbing should be constrained within the component. Maybe worth considering a generic higher order utility component for this?

**Comment by mtias** [comment]
Closing as https://github.com/WordPress/gutenberg/pull/6987 was implemented.
